### PR TITLE
Adding possibility to send some alerts to customers via mailgun

### PR DESCRIPTION
**Pull request by wejdross**
## Summary

This adds the possibility to send alerts triggered by the created instance to the customer via mailgun.
This PR will add the required smtp secret for mailgun to the cluster, which is then used by the alertmanager config of that instance to send alerts to the configured e-mail address in the claim.
The alertmanager config itself is created using composition functions and is implemented here: https://github.com/vshn/appcat/pull/21

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: class/appcat.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ parameters:
   kapitan:
     dependencies:
       - type: git
-        output_path: dependencies/appcat/manifests/${appcat:images:appcat:tag}
+        output_path: dependencies/appcat/manifests/mailgun-alerting
         source: https://github.com/vshn/appcat.git
-        ref: ${appcat:images:appcat:tag}
+        ref: mailgun-alerting
```
Comment (Kidswiss) [inline class/appcat.yml:7]: You probably want to change this back.

File: class/appcat.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ parameters:
   kapitan:
     dependencies:
       - type: git
-        output_path: dependencies/appcat/manifests/${appcat:images:appcat:tag}
+        output_path: dependencies/appcat/manifests/mailgun-alerting
```
Comment (Kidswiss) [inline class/appcat.yml:5]: You probably want to change this back.

File: class/defaults.yml [hunk 3]
```diff
@@ -28,7 +28,7 @@ parameters:
       appcat:
         registry: ghcr.io
         repository: vshn/appcat
-        tag: v4.0.0
+        tag: mailgun-alerting
```
Comment (Kidswiss) [inline class/defaults.yml:31]: You probably want to change this back.

File: component/main.jsonnet [hunk 4]
```diff
@@ -66,12 +68,34 @@ local readServices = kube.ClusterRole('appcat:services:read') + {
   ],
 };
 
-{
-  '10_clusterrole_view': xrdBrowseRole,
-  [if isOpenshift then '10_clusterrole_finalizer']: finalizerRole,
-  '10_clusterrole_services_read': readServices,
+// adding namespace for syn-appcat
+local ns = kube.Namespace('syn-appcat');
+// adding missing namespace for slos
```
Comment (Kidswiss) [inline component/main.jsonnet:73]: The SLI part of the component deploys this one already. Why did you add it again here?

File: component/main.jsonnet [hunk 5]
```diff
@@ -66,12 +68,34 @@ local readServices = kube.ClusterRole('appcat:services:read') + {
...
+local slos_namespace = kube.Namespace('appcat-slos');
+
+// add secret
+
+local secret = kube.Secret('mailgun-smtp-credentials') {
+  metadata+: {
+    namespace: 'syn-appcat',
+  },
+  stringData: {
+    password: pgParams.mailgun_smtp_password,
```
Comment (Kidswiss) [inline component/main.jsonnet:83]: There doesn't seem to be a vault reference in the `defaults.yaml` where does this come from?

File: component/main.jsonnet [hunk 6]
```diff
@@ -66,12 +68,34 @@ local readServices = kube.ClusterRole('appcat:services:read') + {
...
+  },
+};
+
+if params.services.vshn.enabled && pgParams.enabled then
```
Comment (Kidswiss) [inline component/main.jsonnet:87]: This `if` seems to remove those files clusterrole files when generating the golden images. That will probably break a lot of stuff, please change this if statement so that it only applies to the mailgun secret.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
This removes some necessary clusterroles from the golden master, please fix that.

Also please add a summary to your PR description for the changelog.

File: component/main.jsonnet (hunk 4 above)
Comment (wejdross) [reply component/main.jsonnet:73]: because when You setup fresh kind cluster, then namespace is never created and our makefile returns error

File: component/main.jsonnet (hunk 5 above)
Comment (wejdross) [reply component/main.jsonnet:83]: That;s good hint, I'll move it a little bit up in hierarchy - I wanted to add valut values to syn, because each cluster has it's own secret - should it be rather here?

File: component/main.jsonnet (hunk 4 above)
Comment (Kidswiss) [reply component/main.jsonnet:73]: What target fails for you? The `vshnpostgresql` target works for me without error.

But I would still remove it here, or we will have problems with argoCD once we deploy stuff. There should be other means to solve your issue.

File: component/main.jsonnet (hunk 5 above)
Comment (Kidswiss) [reply component/main.jsonnet:83]: If you put the values into the vault you still need a reference in the component.

You can look at other vault values in the component to see how they are handled. For example this here: https://github.com/vshn/component-appcat/blob/master/class/defaults.yml#L125

Also add some documentation how we get those secrets, so everyone could install AppCat on a new cluster.

File: component/main.jsonnet (hunk 6 above)
Comment (Kidswiss) [reply component/main.jsonnet]: This still removes some necessary files it seems.

File: class/defaults.yml [hunk 7]
```diff
@@ -151,6 +151,7 @@ parameters:
       vshn:
         enabled: false
         secretNamespace: ${crossplane:namespace}
+        mailgun_smtp_password: "?{vaultkv:${cluster:tenant}/${cluster:name}/appcat/mailgun_smpt_password}"
```
Comment (Kidswiss) [inline class/defaults.yml:154]: We should probably document where this smtp password comes from.

Also, if we only have one password for mailgun, then maybe we should not store it in a cluster specific place? Then the manual work to install appcat will be a bit less.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Just one thing, the rest LGTM

File: class/defaults.yml (hunk 7 above)
Comment (zugao) [inline class/defaults.yml:154]: Is there a reason for not using camel case here?

File: component/main.jsonnet [hunk 8]
```diff
@@ -66,10 +68,23 @@ local readServices = kube.ClusterRole('appcat:services:read') + {
   ],
 };
 
+// adding namespace for syn-appcat
+local ns = kube.Namespace('syn-appcat');
```
Comment (zugao) [inline component/main.jsonnet:72]: We should probably need to make it configurable in defaults like any other namespace.

**Review by zugao — COMMENTED** [review]
Some things to clarify

File: class/defaults.yml (hunk 7 above)
Comment (TheBigLee) [reply class/defaults.yml]: Fixed

File: component/main.jsonnet (hunk 8 above)
Comment (TheBigLee) [reply component/main.jsonnet]: Fixed

**Review by Kidswiss — APPROVED** [review]

**Review by Kidswiss — APPROVED** [review]

File: component/main.jsonnet [hunk 9]
```diff
@@ -66,10 +68,23 @@ local readServices = kube.ClusterRole('appcat:services:read') + {
...
+local ns = kube.Namespace(params.namespace);
+
+local secret = kube.Secret(params.services.vshn.emailAlerting.secretName) {
+  metadata+: {
+    namespace: params.services.vshn.emailAlerting.secretNamespace,
+  },
+  stringData: {
+    password: params.services.vshn.emailAlerting.smtpPassword,
+  },
+};
```
Comment (glrf) [inline component/main.jsonnet:81]: Nitpick: I'd prefer if we'd only deploy the secret if we actually need it. I.e. if VSHN services are enabled.

Comment (TheBigLee) [reply component/main.jsonnet:81]: Fixed
